go.mod

```diff
@@ -1,14 +1,61 @@
 module github.com/hashicorp/vault-lambda-extension
 
-go 1.15
+go 1.17
```

Comment:
Does this gate the format update below where direct and indirect dependencies are split up? This setting only specifies the minimum version of Go required to build the module, so I'm surprised to see it updated, but I don't have any problem with enforcing recent versions, especially as this is an application and not a library.

Reply:
Yeah exactly, running `go mod tidy` after updating this line to 1.17 produces this new format for go.mod, I think to support [module graph pruning](https://go.dev/doc/go1.17#go-command).

I went ahead and bumped it since as you mentioned this is an application and not a library that other applications would be expected to import.